packages/cli/src/create/data-set/DataSet.handler.ts

```diff
@@ -11,12 +11,69 @@
 
 import type { IHandlerParameters } from "@zowe/imperative";
 import type { ZSshClient, ds } from "zowe-native-proto-sdk";
+import type { DatasetAttributes } from "zowe-native-proto-sdk";
 import { SshBaseHandler } from "../../SshBaseHandler";
+import { CreateDefaults } from "../CreateDefaults";
 
 export default class CreateDatasetHandler extends SshBaseHandler {
     public async processWithClient(params: IHandlerParameters, client: ZSshClient): Promise<ds.CreateDatasetResponse> {
         const dsname = params.arguments.name;
-        const response = await client.ds.createDataset({ dsname, dstype: "default" });
+        let attributes: Partial<DatasetAttributes> = {};
+
+        const attrKeys: (keyof DatasetAttributes)[] = [
+            "alcunit",
+            "blksize",
+            "dirblk",
+            "dsorg",
+            "primary",
+            "recfm",
+            "lrecl",
+            "dataclass",
+            "unit",
+            "dsntype",
+            "mgntclass",
+            "dsname",
+            "avgblk",
+            "secondary",
+            "size",
+            "storclass",
+            "vol",
+        ];
+
+        switch (params.arguments.template?.toLocaleUpperCase()) {
+            case "PARTITIONED":
+                attributes = CreateDefaults.DATA_SET.PARTITIONED;
+                break;
+            case "SEQUENTIAL":
+                attributes = CreateDefaults.DATA_SET.SEQUENTIAL;
+                break;
+            case "DEFAULT":
+                attributes = CreateDefaults.DATA_SET.SEQUENTIAL;
+                break;
+            case "CLASSIC":
+                attributes = CreateDefaults.DATA_SET.CLASSIC;
+                break;
+            case "C":
+                attributes = CreateDefaults.DATA_SET.C;
+                break;
+            case "BINARY":
+                attributes = CreateDefaults.DATA_SET.BINARY;
+                break;
+            default:
+                break;
+        }
+
+        const args = params.arguments as Record<string, unknown>;
+
+        for (const key of attrKeys) {
+            const value = args[key];
+            if (value !== undefined) {
+                // biome-ignore lint/suspicious/noExplicitAny: as any required for attribute mapping
+                (attributes as any)[key] = value;
+            }
+        }
```

Comment:
(nitpick) The `as any` cast feels a bit weird to me but after testing some alternatives in the editor, I can see why its necessary when the attributes are applied this way.

However, can't we just spread `params.arguments` onto attributes? Since we only check keys that are in `keyof DatasetAttributes`, we can just re-assign attributes to contain both the existing & new properties, avoiding the cast:

```suggestion
        for (const key of attrKeys) {
            const value = args[key];
            if (value !== undefined) {
                attributes = { ...attributes, [key]: value as any };
            }
        }
```

Reply:
To take it a step further, we can replace the loop with the functional equivalent `attrKeys.reduce`, but definitely not required 😅 

With reduce, you can specify `attributes` as the initial value. If value is `undefined` we can return the existing attributes object. Otherwise we can return `{ ...attributes, [key]: value }` to add onto the attributes object.